application.go

```diff
@@ -122,10 +122,11 @@ func (r *Application) AllTaskRunning() bool {
 	if r.Tasks == nil {
 		return false
 	}
-	if r.TasksRunning == r.Instances {
-		return true
+	if r.HasHealthChecks() {
+		return r.TasksRunning == r.Instances && r.TasksHealthy == r.Instances
+	} else {
+		return r.TasksRunning == r.Instances
```

Comment:
We can drop the `else` here and directly return the boolean expression as we're already returning in the `if` case. Or, if you agree to my reasoning above, directly go with

```
return r.TasksHealthy == r.Instances
```